plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/util/MessageDialogUtil.java

```diff
@@ -0,0 +1,39 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Red Hat, Inc. Distributed under license by Red Hat, Inc.
+ * All rights reserved. This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution, and is
+ * available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors: Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.openshift.internal.ui.util;
+
+import org.eclipse.jface.dialogs.MessageDialog;
+import org.eclipse.swt.widgets.Display;
+
+/**
+ * Remove the need to remember to sync the UI thread
+ * @author jeff.cantrill
+ *
+ */
+public class MessageDialogUtil {
+	
+	public static void openInformation(String title, String message) {
+		openDialog(MessageDialog.INFORMATION, title, message);
+	}
+	
+	public static void openError(String title, String message) {
+		openDialog(MessageDialog.ERROR, title, message);
+	}
+	
+	private static void openDialog(final int severity, final String title, final String message) {
+		final Display display = Display.getDefault();
+		Display.getDefault().syncExec(new Runnable() {
```

Comment:
Depending on the context in which this is called this can lead to hidden dialog with focus (since you're getting the default display) or even dialogs popping up in the 1st monitor where the user moved Eclipse to the secondary monitor - 4 lines below you get the active shell from the default display. Better would be to get the display from current workbench or current wizard/dialog.